crates/stack/src/packet_conv.rs

```diff
@@ -0,0 +1,106 @@
+use crate::packet::{EtherField, IpPacket, PeekPacket};
+use crate::Error;
+
+pub fn packet_ip_wrap_to_ether(
+    frame: &[u8],
+    src_mac: Option<&[u8; 6]>,
+    dst_mac: Option<&[u8; 6]>,
+) -> Result<Vec<u8>, Error> {
+    if frame.is_empty() {
+        return Err(Error::Other(
+            "Error when wrapping IP packet: Empty packet".into(),
+        ));
+    }
+    if let Err(err) = IpPacket::peek(frame) {
+        return Err(Error::PacketMalformed(format!(
+            "Error when wrapping IP packet {err}"
+        )));
+    }
+
+    let mut eth_packet = vec![0u8; frame.len() + 14];
+    if let Some(dst_mac) = dst_mac {
+        eth_packet[EtherField::DST_MAC].copy_from_slice(dst_mac);
+    } else {
+        const DEFAULT_DST_MAC: &[u8; 6] = &[0x02, 0x02, 0x02, 0x02, 0x02, 0x02];
+        eth_packet[EtherField::DST_MAC].copy_from_slice(DEFAULT_DST_MAC);
+    }
+    if let Some(src_mac) = src_mac {
+        eth_packet[EtherField::SRC_MAC].copy_from_slice(src_mac);
+    } else {
+        const DEFAULT_SRC_MAC: &[u8; 6] = &[0x01, 0x01, 0x01, 0x01, 0x01, 0x01];
```

Comment:
MAC addresses starting with `0x01` are not reserved for private use.
It would be better to use one that starts with `0x02` as for `DEFAULT_DST_MAC`.

```
There are 4 ranges of Locally Administered Address Ranges that can be used on a local network:

x2-xx-xx-xx-xx-xx
x6-xx-xx-xx-xx-xx
xA-xx-xx-xx-xx-xx
xE-xx-xx-xx-xx-xx
```

Reply:
What Mac addresses do you propose for default?
are 22-22-22-22-22-22 and 32-32-32-32-32-32 ok?

Reply:
Ok, fixed defaults to 
        const DEFAULT_DST_MAC: &[u8; 6] = &[0x32, 0x32, 0x32, 0x32, 0x32, 0x32];
        const DEFAULT_SRC_MAC: &[u8; 6] = &[0x22, 0x22, 0x22, 0x22, 0x22, 0x22];